packages/flutter/lib/src/widgets/page_storage.dart

```diff
@@ -129,7 +111,121 @@ class PageStorageBucket {
   }
 }
 
-/// A widget that establishes a page storage bucket for this widget subtree.
+/// Establish a subtree in which widgets can opt into persisting states after
+/// being destroyed.
+/// 
+/// [PageStorage] is used to save and restore values that can outlive the widget
+/// For example, when multiple pages are grouped in tabs, when a page is
+/// switched out, its widget is destroyed and its state is lost. By adding a
+/// [PageStorage] at the root and adding a [PageStorageKey] to each page, the
+/// page's state will be stored in its closest ancestor [PageStorage], and
+/// restored when it's switched back.
+/// Usually you don't need to explicitly use a [PageStorage], since it's already
+/// included in routes.
+/// [PageStorageKey] is used by [Scrollable] if
+/// `keepScrollOffset` is enabled to save their [ScrollPosition]s.
+/// {@tool snippet}
+/// This sample shows how to explicitly use a [PageStorage] to
+/// store the states of its children pages. Each page includes a scrollable
+/// list, whose position is preserved when switching between the tabs thanks to
+/// the help of [PageStorageKey].
+///
+/// ```dart
+/// void main() => runApp(MyApp());
+///
+/// class MyApp extends StatelessWidget {
+///   @override
+///   Widget build(BuildContext context) {
+///     return MaterialApp(
+///       title: 'Flutter Demo',
+///       theme: ThemeData(
+///         primarySwatch: Colors.blue,
+///       ),
+///       home: MyHomePage(),
+///     );
+///   }
+/// }
+///
+/// class MyHomePage extends StatefulWidget {
+///   @override
+///   _MyHomePageState createState() => _MyHomePageState();
+/// }
+///
+/// class _MyHomePageState extends State<MyHomePage> {
+///   static final Key keyOne = PageStorageKey('pageOne');
+///   static final Key keyTwo = PageStorageKey('pageTwo');
+///   final PageStorageBucket bucket = PageStorageBucket();
+///   static final ColorBoxPage one = ColorBoxPage(
+///     key: keyOne,
+///   );
+///   static final ColorBoxPage two = ColorBoxPage(
+///     key: keyTwo,
+///   );
+///   List<Widget> pages = [one, two];
+///   Widget currentPage = one;
+///   int currentTab = 0;
+///
+///   @override
+///   Widget build(BuildContext context) {
+///     return Scaffold(
+///       appBar: AppBar(
+///         title: Text("Persistance Example"),
+///       ),
+///       body: PageStorage(
+///         child: currentPage,
+///         bucket: bucket,
+///       ),
+///       bottomNavigationBar: BottomNavigationBar(
+///         currentIndex: currentTab,
+///         onTap: (int index) {
+///           setState(() {
+///             currentTab = index;
+///             currentPage = pages[index];
+///           });
+///         },
+///         items: <BottomNavigationBarItem>[
+///           BottomNavigationBarItem(
+///             icon: Icon(Icons.home),
```

Comment:
The `icon` is unnecessary. Let's make this sample more concise :)

Reply:
Since I delete `icon`, the example raises an error.
`'package:flutter/src/widgets/bottom_navigation_bar_item.dart': Failed assertion: line 31 pos 15: 'icon != null': is not true.`

I think `icon` a necessary parameter.  `Title` is optional.

Reply:
Sorry. I try to delete the `title`. The app throw an error.
Thus `title` is necessary, too.